### PR TITLE
Keerthik master changes

### PR DESCRIPTION
**Pull request by cryptomail**
This PR supersedes https://github.com/facetoe/zenpy/pull/637

Addresses Issue https://github.com/facetoe/zenpy/issues/633 and adds some other minor improvements

3 main changes

added content_tag specification, api, mapping, endpoint for new [content tags api](https://developer.zendesk.com/api-reference/help_center/help-center-api/content_tags/).
updated CONTRIBUTING.md to be more descriptive for adding new objects (as I learned how to do it)
regenerated classes
1 minor change

in setup.cfg, changed description-file -> description_file because setuptools complained
EDIT: one additional small but significant change is gen_classes.py execution is wrapped in if __name__  == "__main__": so that modern python multiprocessing.pool doesn't complain and fail. This class is only applicable to package development.

### CODE REVIEW

**Comment by cryptomail** [comment]
@keerthik This needs tests.  I'll try to get to them today.  This API needs a bit of exercising ^_^ before we just push / merge.
Screenshot 2024-08-19 at 1 47 17 PM

**Comment by cryptomail** [comment]
@keerthik how's this lookin?

**Comment by cryptomail** [comment]
@keerthik This is looking `almost` OK.  @Ptr314 is currently looking at some deprecation warnings related to Python 3.12 which is not officially supported (yet) but probably? should be.

**Comment by cryptomail** [comment]
With that said, I'm thinking of EOL'ing our 3.8 build and support and *ADDING* 3.12 ^_^ after @Ptr314 gets back to us on a solution.

**Comment by keerthik** [comment]
@cryptomail I am not familiar with the testing specifications involved so I am probably not qualified to comment on it, but it looks like it covers the relevant cases.

It looks like your changes preserve all the important modifications from my commits to enable content tag support (changing the datetime dependencies looks like a positive long-term change I had not included), so it looks good to me. 

Looking forward to the next release including these changes so I can use python 3.12 and interact with content tags without relying on my local forked pypackage.

**Comment by cryptomail** [comment]
> @cryptomail I am not familiar with the testing specifications involved so I am probably not qualified to comment on it, but it looks like it covers the relevant cases.
> 
> It looks like your changes preserve all the important modifications from my commits to enable content tag support (changing the datetime dependencies looks like a positive long-term change I had not included), so it looks good to me.
> 
> Looking forward to the next release including these changes so I can use python 3.12 and interact with content tags without relying on my local forked pypackage.

We are here for you, we'll do our level-best to ensure we modernize appropriately and in a timely fashion.  We'll get you to mainline in a minute...Thanks for holding <3

**Comment by cryptomail** [comment]
> So, we had two deprecaton warnings for 3.12 - one for betamax, what we cannot change, and one in a test case which I've fixed. 3.8 is okay for both and for my new fix too. To ignore warnings in 3.12 just run tests like this: "pytest -Wignore" until they fix betamax. I've not tested on a live account yet, it will take a little time.
> 
> Details about the root of the problem: https://blog.miguelgrinberg.com/post/it-s-time-for-a-change-datetime-utcnow-is-now-deprecated.
> 
> @cryptomail I think we can't include 3.12 to our CI/CD pipeline until betamax is ready for it. The last version is 0.9.0 and it still has this issue.

@Ptr314 Yikes!  This isn't the best of news.....BUT we'll do our best as things unfold.....
Worst case is we may have to lean in (in this case pretty hard) to BetaMax le sigh.....)

**Comment by Ptr314** [comment]
@keerthik The library itself doesn't have any problems with 3.12, just our testing procedures have to be fixed.

**Comment by cryptomail** [comment]
> @keerthik The library itself doesn't have any problems with 3.12, just our testing procedures have to be fixed.

Oh you're saying we just can change things on our side....I see...well....in that case it's a bit of local energy on our part haha :)

File: tests/test_api/test_other_zendesk_objects.py [hunk 1]
```diff
@@ -34,11 +34,13 @@
...
 )
 from zenpy.lib import util
-from datetime import datetime
+from datetime import datetime, timezone
 from unittest import TestCase
+
 class DateTimeTest(TestCase):
     def test_datetime_import(self):
-        util.to_unix_ts(datetime.utcnow())
+        util.to_unix_ts(datetime.now(timezone.utc))
```
Comment (cryptomail) [inline tests/test_api/test_other_zendesk_objects.py:42]: 👨‍🍳 💋

**Comment by cryptomail** [comment]
This is pretty hopeful: https://github.com/betamaxpy/betamax/pull/210/files
But they haven't released 0.91 YET!!!

**Comment by Ptr314** [comment]
> Oh you're saying we just can change things on our side....I see...well....in that case it's a bit of local energy on our part haha :)

I think we can just relax right now and wait until betamax gurus do their work. 

Just tried live testing with 3.8 and 3.12, excluding some specific tests like webhooks etc. that require some instance dependent data. My conclusion is we can make a new release with this PR.

**Comment by cryptomail** [comment]
confirmed: zero warnings from HEAD hahaha, they just need a release :)
Screenshot 2024-08-20 at 1 08 52 PM

**Pull request merged**